### PR TITLE
feat: BlockaidBannerAlert update Failed type false positive support

### LINKED ISSUE #22711: [Feat]: Support Blockaid Reporting False Positives Failed result type
### What is this about?

Currently, when a transaction fails, a user can submit a false positive report. We want to update the parameters we pass based on specs from Blockaid to fully support this. 

Parameters to update when result_type is `Failed: 
- classification → 'error'
- result_type → 'Error'

Please see [internal thread](https://consensys.slack.com/archives/C05HZHDS89Z/p1706541505600509?thread_ts=1705393852.556679&cid=C05HZHDS89Z) for more details

### Scenario

_No response_

### Design

_No response_

### Technical Details

_No response_

### Threat Modeling Framework

_No response_

### Acceptance Criteria

_No response_

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by digiwand**
## **Description**

See [issue](https://github.com/MetaMask/metamask-extension/issues/22711) for details

note: we will need to resolve merge conflicts with https://github.com/MetaMask/metamask-extension/pull/22625 

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/22711

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by segun — COMMENTED** [review]
This looks good, only address the comment by @jpuri

File: ui/components/app/security-provider-banner-alert/blockaid-banner-alert/blockaid-banner-alert.js [hunk 1]
```diff
@@ -97,15 +97,16 @@ function BlockaidBannerAlert({ txData, ...props }) {
...
     jsonRpcMethod: type,
     jsonRpcParams: JSON.stringify(txParams ?? msgParams),
     blockNumber: block,
     chain: NETWORK_TO_NAME_MAP[chainId],
-    classification: reason,
+    classification: isFailedResultType ? 'error' : reason,
     blockaidVersion: BlockaidPackage.version,
-    resultType,
+    resultType: isFailedResultType ? 'Error' : resultType,
     reproduce: JSON.stringify(features),
```
[the remark this answers is not in the document]
Comment (digiwand) [reply ui/components/app/security-provider-banner-alert/blockaid-banner-alert/blockaid-banner-alert.js:110]: interesting.  looked at the code and remembered that "Failed" was defined by us. Should we create a follow-up PR to replace all instances of "Failed" with "Error" now that this result_type, "Error",  has been added by Blockaid? 
cc: @jpuri @segun

[the remark this answers is not in the document]
Comment (digiwand) [reply ui/components/app/security-provider-banner-alert/blockaid-banner-alert/blockaid-banner-alert.js:110]: or I can update this PR to do this

[the remark this answers is not in the document]
Comment (digiwand) [reply ui/components/app/security-provider-banner-alert/blockaid-banner-alert/blockaid-banner-alert.js:110]: added Error ResultType constant here [6127a82](https://github.com/MetaMask/metamask-extension/pull/22742/commits/6127a82854c5d1ae72a29ab24a8f3eebb70d9525). 

to replace the "Failed" instances, we can also add the change to this PR

**Review by jpuri — APPROVED** [review]

**Review by segun — APPROVED** [review]

**Pull request merged**